consensus/README.md

```diff
@@ -140,6 +140,222 @@ graph TB
   J --> note2
 ```
 
+## Synchronization between the consensus processes
+
+The consensus module currently depends on the `PaceMaker`, `StateSync`, `LeaderElection`, `Networking`.
+It has a bootstrapping state where it:
+  * Initializes connections to the network through a bootstrap node
+  * Gets the latest block then switches to one of the mutually exclusive modes: `sync` or `consensus`
+  * Keeps an updated current height (the greatest block height seen on the network)
+
+```mermaid
+sequenceDiagram
+participant Consensus
+participant StateSync
+Note over Node: This embeds Networking, FSM
+participant Node
+participant Network
+Node->>Node:Start
+Node->>Network:Join
+loop
+  Node->>Network:RequestCurrentHeight
+  Network-->>Node: 
+  Node->>Node:SwitchMode
+end
+alt SyncMode
+  par
+    Node->>Consensus:Pause
+  and
+    Node->>StateSync:Start
+    Note over Node,StateSync: Details omitted for another diagram
+    StateSync-->>Node:SyncDone
+  end
+else ConsensusMode
+  par
+    Node->>StateSync:Pause
+  and
+    Node->>Consensus:Start
+    Consensus-->>Node:OutOfSync
+  end
+end
+```
+
+### StateSync Mode
+
+In this mode the node is behind the latest known-to-network height and will try to catchup by downloading then applying the downloaded blocks (sequentially) to its local state.
+
+The `download` and `apply` routines may run in parallel, but `apply` may be blocked by the former if the needed block is missing, it will wait until the needed blocks are downloaded.
+
+#### Download routine
+
+* The `download` routine is alive as long as `sync` mode is on
+* It checks in its persistence for the latest downloaded block and tries to get and add to its persistence all the blocks; up to the network current height
+* After downloading, and before inserting the block, basic (stateless) verification is made to the block
+* A downloaded and inserted block is a structurally valid block but should by no mean considered valid w.r.t. its validators signatures or transactions within
+
+```mermaid
+sequenceDiagram
+participant Persistence
+Note over Node: (Download routine)
+participant Node
+participant Network
+par
+  loop
+  Note left of Network: Constantly asking for the highest<br /> known block from network
```

Comment:
Consider using activations: https://mermaid.js.org/syntax/sequenceDiagram.html#activations

Reply:
How would I use it?

Reply:
Similar to my comment from yesterday here: https://github.com/pokt-network/pocket/pull/951#discussion_r1280001737

![Screenshot 2023-08-01 at 11 47 40 AM](https://github.com/pokt-network/pocket/assets/1892194/6551e4ec-47c9-49a4-8897-f9c2882fd8d8)

But I noticed that you did not include it in the edits.

Let me know if:
1. Did it not make sense?
2. If you disagree?
3. If you just need more details/clarification of why I think it's appropriate